data-prepper-plugins/kafka-connect-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafkaconnect/source/KafkaConnectSource.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.kafkaconnect.source;
+
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.configuration.PipelineDescription;
+import org.opensearch.dataprepper.model.plugin.kafka.AuthConfig;
+import org.opensearch.dataprepper.model.plugin.kafka.AwsConfig;
+import org.opensearch.dataprepper.model.plugin.kafka.EncryptionConfig;
+import org.opensearch.dataprepper.model.plugin.kafka.UsesKafkaClusterConfig;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.plugins.kafka.util.KafkaSourceSecurityConfigurer;
+import org.opensearch.dataprepper.plugins.kafkaconnect.util.Connector;
+import org.opensearch.dataprepper.plugins.kafkaconnect.configuration.KafkaConnectSourceConfig;
+import org.opensearch.dataprepper.plugins.kafkaconnect.configuration.WorkerProperties;
+import org.opensearch.dataprepper.plugins.kafkaconnect.util.KafkaConnect;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * The starting point of the debezium-source.
+ * The debezium engine is configured and runs async here.
+ */
+
+@SuppressWarnings("deprecation")
+@DataPrepperPlugin(name = "kafka_connect", pluginType = Source.class, pluginConfigurationType = KafkaConnectSourceConfig.class)
```

Comment:
Please see this comment:

https://github.com/opensearch-project/data-prepper/issues/3233#issuecomment-1699806916

As noted there, we should have different sources for each of the databases: Postgresql, MySQL, MongoDB.

A good way to solve this would be to add a Kafka Connect extensions. See the `aws-plugin-api` and `aws-plugin` projects for examples.

You could have a `kafka-connect-api` project which provides a basic API. Then, the `mysql-source` project would depend on it. (Similar for `postgres-source`, and `mongodb-source`).

The implementation can be an extensions in a project such as `kakfa-connect-extension`.